### PR TITLE
Fix printing of adb logcat

### PR DESCRIPTION
**Pull request by rosen-vladimirov**
Also add quotes around path to adb, so even when the path contains spaces, the command will be executed without troubles.

Fix checking for javac version to use JAVA_HOME in case it is defined.

### CODE REVIEW

**Comment by justcodebuilduser** [comment]
:heart:

File: mobile/android/logcat-helper.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export class LogcatHelper implements Mobile.ILogcatHelper {
 	public start(deviceIdentifier: string): void {
 		if(deviceIdentifier) {
 			let adbPath = this.$staticConfig.getAdbFilePath().wait();
-			this.$childProcess.exec(util.format("%s -s %s logcat -c", adbPath, deviceIdentifier)).wait(); // remove cached logs
+			this.$childProcess.exec(util.format('"%s" -s "%s" logcat -c', adbPath, deviceIdentifier)).wait(); // remove cached logs
```
Comment (ligaz) [inline mobile/android/logcat-helper.ts]: You can use template strings here.

**Comment by justcodebuilduser** [comment]
:heart:

File: mobile/android/logcat-helper.ts [hunk 2]
```diff
@@ -13,7 +12,8 @@ export class LogcatHelper implements Mobile.ILogcatHelper {
 	public start(deviceIdentifier: string): void {
 		if(deviceIdentifier) {
 			let adbPath = this.$staticConfig.getAdbFilePath().wait();
-			this.$childProcess.exec(util.format("%s -s %s logcat -c", adbPath, deviceIdentifier)).wait(); // remove cached logs
+			// remove cached logs:
+			this.$childProcess.exec(`"${adbPath}" -s "${deviceIdentifier}" logcat -c`).wait();
```
Comment (teobugslayer) [inline mobile/android/logcat-helper.ts]: can we use spawn / execFile instead?

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by teobugslayer** [comment]
:+1:

**Comment by justcodebuilduser** [comment]
:heart:

**Pull request merged**